internal/oci/runtime_oci.go

```diff
@@ -898,73 +899,85 @@ func (r *runtimeOCI) AttachContainer(c *Container, inputStream io.Reader, output
 	return nil
 }
 
-// PortForwardContainer forwards the specified port provides statistics of a container.
-func (r *runtimeOCI) PortForwardContainer(c *Container, port int32, stream io.ReadWriter) error {
-	emptyStreamOnError := true
-	defer func() {
-		if emptyStreamOnError {
-			go func() {
-				if _, copyError := pools.Copy(ioutil.Discard, stream); copyError != nil {
-					logrus.Errorf("error closing port forward stream after other error: %v", copyError)
-				}
-			}()
+// PortForwardContainer forwards the specified port into the provided container.
+func (r *runtimeOCI) PortForwardContainer(
+	ctx context.Context, c *Container, netNsPath, ip string,
+	port int32, stream io.ReadWriteCloser,
+) error {
+	log.Infof(ctx,
+		"Starting port forward for %s in network namespace %s", c.ID(), netNsPath,
+	)
+
+	// Adapted reference implementation:
+	// https://github.com/containerd/cri/blob/8c366d/pkg/server/sandbox_portforward_unix.go#L65-L120
+	if err := ns.WithNetNSPath(netNsPath, func(_ ns.NetNS) error {
+		defer stream.Close()
+
+		conn, err := net.Dial("tcp4", fmt.Sprintf("%s:%d", ip, port))
```

Comment:
for being IPv6 and IPv4 compatible you hast to change this to:

`net.Dial("tcp", net.JoinHostPort(ip, strconv.Itoa(port)))`

Reply:
Yes I changed it as suggested.